x-pack/plugins/apm/server/routes/alerts/rule_types/error_count/register_error_count_rule_type.test.ts

```diff
@@ -584,4 +584,117 @@ describe('Error count alert', () => {
       alertDetailsUrl: 'mockedAlertsLocator > getLocation',
     });
   });
+
+  it('sends alert when rule is configured with group by on error.grouping_key and error.grouping_name', async () => {
+    const { services, dependencies, executor, scheduleActions } =
+      createRuleTypeMocks();
+
+    registerErrorCountRuleType(dependencies);
+
+    const params = {
+      threshold: 2,
+      windowSize: 5,
+      windowUnit: 'm',
+      groupBy: [
+        'service.name',
+        'service.environment',
+        'error.grouping_key',
+        'error.grouping_name',
+      ],
+    };
+
+    services.scopedClusterClient.asCurrentUser.search.mockResponse({
+      hits: {
+        hits: [],
+        total: {
+          relation: 'eq',
+          value: 2,
+        },
+      },
+      aggregations: {
+        error_counts: {
+          buckets: [
+            {
+              key: ['foo', 'env-foo', 'error-key-foo', 'error-name-foo'],
+              doc_count: 5,
+            },
+            {
+              key: ['foo', 'env-foo-2', 'error-key-foo-2', 'error-name-foo2'],
+              doc_count: 4,
+            },
+            {
+              key: ['bar', 'env-bar', 'error-key-bar', 'error-name-bar'],
+              doc_count: 3,
+            },
+            {
+              key: ['bar', 'env-bar-2', 'error-key-bar-2', 'error-name-bar2'],
+              doc_count: 1,
+            },
+          ],
+        },
+      },
+      took: 0,
+      timed_out: false,
+      _shards: {
+        failed: 0,
+        skipped: 0,
+        successful: 1,
+        total: 1,
+      },
+    });
```

Comment:
It would be better to have these as API tests if that's possible?

Reply:
@sqren I have added API tests in: `x-pack/test/apm_api_integration/tests/alerts/error_count_threshold.spec.ts`. Did you mean to add tests for any specific scenario or are these enough?